platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/DefaultConvention.java

```diff
@@ -54,7 +54,7 @@ public class DefaultConvention implements org.gradle.api.plugins.Convention, Ext
     private final InstanceGenerator instanceGenerator;
 
     private Map<String, Object> plugins;
-    private Map<Object, BeanDynamicObject> dynamicObjects;
+    private IdentityHashMap<Object, BeanDynamicObject> dynamicObjects;
```

Comment:
💭  ErrorProne [insists](https://errorprone.info/bugpattern/IdentityHashMapUsage) that IndentityHashMap shouldn't be mixed with Map because of a different contract with regard to hashCode/equals.

Fixing it doesn't look too ugly anyway.